src/main/java/hudson/remoting/Util.java

```diff
@@ -247,4 +251,26 @@ static InetSocketAddress getResolvedHttpProxyAddress(String host, int port) thro
         }
         return targetAddress;
     }
+
+    @IgnoreJRERequirement @SuppressWarnings("Since15")
+    static void mkdirs(@Nonnull File file) throws IOException {
+        if (file.isDirectory()) return;
+
+        try {
+            Class.forName("java.nio.file.Files");
+            Files.createDirectories(file.toPath());
+            return;
+        } catch (ClassNotFoundException e) {
+            // JDK6
+        } catch (ExceptionInInitializerError e) {
+            // JDK7 on multibyte encoding (http://bugs.java.com/bugdatabase/view_bug.do?bug_id=7050570)
+        }
+
+        // Fallback
+        if (!file.mkdirs()) {
+            if (!file.isDirectory()) {
+                throw new IOException("Directory not created");
```

Comment:
Which directory, the target one? I think that is out of the scope of the method.